Compilation failure on nRF54H20 VPR with UART memory region

**Describe the bug**
Recently, the `uart_nrfx_uarte` driver adopted DMM and exposed a bug with one of the DMM macros. It may cause compilation to fail when building for a target with no data cache, such as the nRF54H20 PPR core.

**To Reproduce**
1. Create `samples/hello_world/app.overlay`:
```
&uart135 {
    memory-regions = <&cpuapp_dma_region>;
};
```
2. Run `west build -p samples/hello_world -b nrf54h20dk//cpuppr`

**Expected behavior**
Build success

**Impact**
Showstopper

**Logs and console output**
```
FAILED: zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_nrfx_uarte.c.obj
/home/gms/ncs/zephyr-sdk-0.16.8/riscv64-zephyr-elf/bin/riscv64-zephyr-elf-gcc -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DNRF54H20_XXAA -DNRF_PPR -DPICOLIBC_LONG_LONG_PRINTF_SCANF -D__LINUX_ERRNO_EXTENSIONS__ -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I/home/gms/ncs/zephyr/samples/hello_world/build/zephyr/include/generated/zephyr -I/home/gms/ncs/zephyr/include -I/home/gms/ncs/zephyr/samples/hello_world/build/zephyr/include/generated -I/home/gms/ncs/zephyr/soc/nordic -I/home/gms/ncs/zephyr/soc/common/riscv-privileged/. -I/home/gms/ncs/zephyr/soc/nordic/nrf54h/. -I/home/gms/ncs/zephyr/soc/nordic/nrf54h/gpd/include -I/home/gms/ncs/zephyr/soc/nordic/common/vpr/. -I/home/gms/ncs/zephyr/soc/nordic/common/. -I/home/gms/ncs/modules/hal/nordic/nrfx -I/home/gms/ncs/modules/hal/nordic/nrfx/drivers/include -I/home/gms/ncs/modules/hal/nordic/nrfx/mdk -I/home/gms/ncs/zephyr/modules/hal_nordic/nrfx/. -isystem /home/gms/ncs/zephyr/lib/libc/common/include -Wshadow -fno-strict-aliasing -Os -imacros /home/gms/ncs/zephyr/samples/hello_world/build/zephyr/include/generated/zephyr/autoconf.h -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mabi=ilp32e -march=rv32emc_zicsr --sysroot=/home/gms/ncs/zephyr-sdk-0.16.8/riscv64-zephyr-elf/riscv64-zephyr-elf -imacros /home/gms/ncs/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/gms/ncs/zephyr/samples/hello_world=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/gms/ncs/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/gms/ncs=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=picolibc.specs -std=c99 -MD -MT zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_nrfx_uarte.c.obj -MF zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_nrfx_uarte.c.obj.d -o zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_nrfx_uarte.c.obj -c /home/gms/ncs/zephyr/drivers/serial/uart_nrfx_uarte.c
In file included from /home/gms/ncs/modules/hal/nordic/nrfx/drivers/nrfx_utils.h:37,
                 from /home/gms/ncs/modules/hal/nordic/nrfx/drivers/nrfx_common.h:44,
                 from /home/gms/ncs/modules/hal/nordic/nrfx/nrfx.h:38,
                 from /home/gms/ncs/modules/hal/nordic/nrfx/hal/nrf_uarte.h:37,
                 from /home/gms/ncs/zephyr/drivers/serial/uart_nrfx_uarte.c:15:
/home/gms/ncs/zephyr/soc/nordic/common/./dmm.h:37:42: error: 'CONFIG_DCACHE_LINE_SIZE' undeclared here (not in a function); did you mean 'CONFIG_TIMESLICE_SIZE'?
   37 |         (DMM_IS_REG_CACHEABLE(node_id) ? CONFIG_DCACHE_LINE_SIZE : sizeof(uint8_t))
```

**Environment:**

 - OS: Linux
 - Toolchain: Zephyr SDK 0.16.8
 - Commit SHA or Version used: v4.0.0-rc2

--- a/soc/nordic/common/dmm.h
+++ b/soc/nordic/common/dmm.h
@@ -23,18 +23,26 @@ extern "C" {
 
 /** @cond INTERNAL_HIDDEN */
 
+#ifdef CONFIG_DCACHE
+
 /* Determine if memory region is cacheable. */
-#define DMM_IS_REG_CACHEABLE(node_id)					 \
-	COND_CODE_1(CONFIG_DCACHE,					 \
-	   (COND_CODE_1(DT_NODE_HAS_PROP(node_id, zephyr_memory_attr),	 \
-	    ((DT_PROP(node_id, zephyr_memory_attr) & DT_MEM_CACHEABLE)), \
-	    (0))), (0))
+#define DMM_IS_REG_CACHEABLE(node_id)					     \
+	COND_CODE_1(DT_NODE_HAS_PROP(node_id, zephyr_memory_attr),	     \
+		((DT_PROP(node_id, zephyr_memory_attr) & DT_MEM_CACHEABLE)), \
+		(0))
 
 /* Determine required alignment of the data buffers in specified memory region.
  * Cache line alignment is required if region is cacheable and data cache is enabled.
  */
 #define DMM_REG_ALIGN_SIZE(node_id) \
 	(DMM_IS_REG_CACHEABLE(node_id) ? CONFIG_DCACHE_LINE_SIZE : sizeof(uint8_t))
+
+#else
+
+#define DMM_IS_REG_CACHEABLE(node_id) 0
+#define DMM_REG_ALIGN_SIZE(node_id) (sizeof(uint8_t))
+
+#endif /* CONFIG_DCACHE */
 
 /* Determine required alignment of the data buffers in memory region
  * associated with specified device node.

